File: grind75/Dec20/973.py
from heapq import heappush, heappop
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> 
List[List[int]]:
        heap = []
        for point in points:
            heappush(heap, [-sqrt(abs(point[0]) ** 2 + abs(point[1] ** 
2)), point[0], point[1]])
            if len(heap) > k:
                heappop(heap)
        return [[p[1], p[2]] for p in heap]

File: grind75/Dec20/973.go
func kClosest(points [][]int, k int) [][]int {
    sort.Slice(points, func(i, j int) bool {
        a := points[i][0]*points[i][0]+ points[i][1]*points[i][1]
        b := points[j][0]*points[j][0]+ points[j][1]*points[j][1]
        return a < b
    })
    return points[:k]
}
